extmod/modtls_mbedtls.c

```diff
@@ -206,6 +189,14 @@ STATIC void ssl_check_async_handshake_failure(mp_obj_ssl_socket_t *sslsock, int
     }
 }
 
+STATIC int ssl_sock_cert_verify(void *ptr, mbedtls_x509_crt *crt, int depth, uint32_t *flags) {
+    mp_obj_ssl_context_t *o = ptr;
+    if (o->handler == mp_const_none) {
+        return 0;
+    }
+    return mp_obj_get_int(mp_call_function_2(o->handler, mp_obj_new_bytearray_by_ref(crt->raw.len, crt->raw.p), MP_OBJ_NEW_SMALL_INT(depth)));
```

Comment:
Actually, thinking about this some more, it's probable even better to use a `memoryview`, because then it can be made read-only.

This is how it's done for `bluetooth.BLE`, in the IRQ callback handler.

Eg:
```c
mp_obj_array_t cert;
mp_obj_memoryview_init(&cert, 'B', 0, crt->raw.len, crt->raw.len);
return mp_obj_get_int(mp_call_function_2(o->handler, MP_OBJ_NEW_FROM_PTR(&cert), MP_OBJ_NEW_SMALL_INT(depth)));
```

Reply:
Thanks for updating with this suggestion.  Looks good now!